### ISSUE TITLE
LocalBusiness type does not recognise `@type: 'LocalBusiness'`

### ISSUE THREAD

**Issue by a-ogilvie** [google/schema-dts#129]
This code reports an error:

```typescript
import { LocalBusiness } from 'schema-dts';

const localBusiness: LocalBusiness = {
  '@context': 'https://schema.org',
  "@id": 'someId',
  '@type': 'LocalBusiness',
  address: { '@type': 'PostalAddress', addressCountry: 'JP' },
  name: 'someName',
};

export default localBusiness;
```

The reported error:

```txt
Type '"LocalBusiness"' is not assignable to type '"TravelAgency"'. ts(2322)
```

![Screenshot 2020-10-23 at 12 16 41](https://user-images.githubusercontent.com/26034704/96952445-d701e480-1529-11eb-949a-1a1d7dd35a7e.png)

**Comment by Eyas** [comment]
The error is misleading, but TypeScript is being tripped up by `"@context"` and excess property checks.

tl;dr: If this is a top-level object, change the type to `WithContext<LocalBusiness>`. Or otherwise drop `@context` (you only need a context at the topmost JSON-LD object or graph).

### Explanation

TypeScript generally has excess property checks to make sure you're not misspelling something, e.g.:

```
type A = { x?: string; y?: string; }
const thing: A = {z: string};
```

while technically `thing` _is_ a subtype of `A`, this is often a sign of mistake, so TS will warn that `z` is wrong.

In schema-dts, `@context` is only supplied in `WithContext<T>`. Unfortunately, instead of straightforwardly complaining about `@context` not being found, TS does it in a very weird way when there's a discriminated union.

**Comment by Eyas** [comment]
I think we can close this, but let me know if you have any issues here!

**Comment by a-ogilvie** [comment]
@Eyas Thank you!! It feels great to have some type safety on my structured data. Thanks to this library, I found a small error in the existing data.

I wonder if the `WithContext` trick could be documented somewhere. I tried Googling my way out of the problem, but I couldn't find any documentation on this error.

**Comment by Eyas** [comment]
That's a good point! `WithContext` is briefly mentioned in the "Root Context" section of the readme (right under usage), but if you're troubleshooting your failure (especially cryptic sounding errors from TS) then you'll have trouble finding `WithContext` as the answer.

Honestly I'm open to feedback. I agree additional documentation is helpful, but I'm not sure what/where. You can always open a PR and fix/add docs as well!
